test/DGmethods/Horizontal_integral_test.jl

```diff
@@ -0,0 +1,98 @@
+using MPI
+using CLIMA
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Grids
+using CLIMA.DGmethods
+using CLIMA.DGmethods.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using CLIMA.Atmos
+using CLIMA.VariableTemplates
+using CLIMA.MoistThermodynamics
+using CLIMA.PlanetParameters
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using CLIMA.VTK
+using CLIMA.Atmos: vars_state, vars_aux
+using DelimitedFiles
+using GPUifyLoops
+using Random
+using CLIMA.IOstrings
+@static if haspkg("CuArrays")
+  using CUDAdrv
+  using CUDAnative
+  using CuArrays
+  CuArrays.allowscalar(false)
+  const ArrayTypes = (CuArray,) 
+else
+  const ArrayTypes = (Array,)
+end
+FT=Float64
+MPI.Initialized() || MPI.Init()
+  mpicomm = MPI.COMM_WORLD
+ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+  loglevel = ll == "DEBUG" ? Logging.Debug :
+    ll == "WARN"  ? Logging.Warn  :
+    ll == "ERROR" ? Logging.Error : Logging.Info
+  logger_stream = MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull
+  global_logger(ConsoleLogger(logger_stream, loglevel))
+@static if haspkg("CUDAnative")
+      device!(MPI.Comm_rank(mpicomm) % length(devices()))
+  end
+for ArrayType in ArrayTypes
+warpfun = (ξ1, ξ2, ξ3) -> begin
+  x1 = ξ1 + (ξ1 - 1/2) * cos(2 * π * ξ2 * ξ3) / 4
+  x2 = ξ2  + (ξ2 - 1/2) * cos(2 * π * ξ2 * ξ3) / 4
+  x3 = ξ3 + ξ1 / 4 + ξ2^2 / 2 + sin(ξ1 * ξ2 * ξ3) + exp(sin(2π * (ξ1 * ξ2 + ξ3)))/20
+  return (x1, x2, x3)
+end
+for N in 4:4
+for Ne in 1:20
```

Comment:
This seems odd to me, why loop over Ne?

Reply:
To see if there is any difference if we increase the number of horizontal elements. Can be removed if unnecessary